Libraries/LibWeb/CSS/CountersSet.cpp

```diff
@@ -55,33 +56,43 @@ Counter& CountersSet::instantiate_a_counter(FlyString const& name, DOM::Abstract
 {
-    if (auto existing_counter = last_counter_with_name(name); existing_counter.has_value()) {
-        existing_counter->value = value;
+    auto existing_counter = last_counter_with_name(name);
+
+    if (!existing_counter.has_value()) {
+        // If there is not currently a counter of the given name on the element, the element instantiates
+        // a new counter of the given name with a starting value of 0 before setting or incrementing its value.
+        // https://drafts.csswg.org/css-lists-3/#valdef-counter-set-counter-name-integer
+        auto& counter = instantiate_a_counter(name, element, false, 0);
+        counter.value = value;
         return;
     }
 
-    // If there is not currently a counter of the given name on the element, the element instantiates
-    // a new counter of the given name with a starting value of 0 before setting or incrementing its value.
-    // https://drafts.csswg.org/css-lists-3/#valdef-counter-set-counter-name-integer
-    auto& counter = instantiate_a_counter(name, element, false, 0);
-    counter.value = value;
+    existing_counter->value = value;
+
+    if (existing_counter->reversed && !existing_counter->is_explicitly_set_reversed_counter) {
+        existing_counter->originating_element.update_initial_value_for_reversed_counter__after_set(name, value.value());
+        existing_counter->is_explicitly_set_reversed_counter = true;
+    }
 }
 
 // https://drafts.csswg.org/css-lists-3/#propdef-counter-increment
 void CountersSet::increment_a_counter(FlyString const& name, DOM::AbstractElement const& element, CounterValue amount)
 {
-    if (auto existing_counter = last_counter_with_name(name); existing_counter.has_value()) {
-        // FIXME: How should we handle existing counters with no value? Can that happen?
-        VERIFY(existing_counter->value.has_value());
-        existing_counter->value->saturating_add(amount.value());
+    auto existing_counter = last_counter_with_name(name);
+
+    if (!existing_counter.has_value()) {
+        // If there is not currently a counter of the given name on the element, the element instantiates
+        // a new counter of the given name with a starting value of 0 before setting or incrementing its value.
+        // https://drafts.csswg.org/css-lists-3/#valdef-counter-set-counter-name-integer
+        auto& counter = instantiate_a_counter(name, element, false, 0);
+        counter.value->saturating_add(amount.value());
         return;
     }
 
-    // If there is not currently a counter of the given name on the element, the element instantiates
-    // a new counter of the given name with a starting value of 0 before setting or incrementing its value.
-    // https://drafts.csswg.org/css-lists-3/#valdef-counter-set-counter-name-integer
-    auto& counter = instantiate_a_counter(name, element, false, 0);
-    counter.value->saturating_add(amount.value());
+    if (!existing_counter->value.has_value())
+        existing_counter->value = 0;
```

Comment:
This is to support "fixing up" the reversed value later, right? It could do with a comment.